src/ast.rs

```diff
@@ -1018,6 +1071,15 @@ impl AstStatement {
             AstStatement::CastStatement { id, .. } => *id,
         }
     }
+
+    /// Returns true if the current statement has a return access.
+    pub fn has_direct_access(&self) -> bool {
```

Comment:
I find this very very specific ... for such a method on the level of AstStatement I would expect it to also return true on an expression-Tree
```
1 + X.%B1
```
I would keep this as a helper function where you need it